platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/MigrationZipFile.java

```diff
@@ -0,0 +1,255 @@
+
+  /**
+   * Opens a MigrationZipFile for reading encrypted entries using default locations for the key and
+   * checksum files
+   *
+   * @param zipPath path to zip file
+   * @throws IOException when there is a problem reading the zip file
+   */
+  public MigrationZipFile(Path zipPath) throws IOException {
+    this(zipPath, getDefaultKeyPathFor(zipPath), getDefaultChecksumPathFor(zipPath));
+  }
+
+  /**
+   * Opens a MigrationZipFile for reading encrypted entries using custom locations for the key and
+   * checksum files
+   *
+   * @param zipPath path to zip file
+   * @param keyPath path to key file
+   * @param checksumPath path to checksum file
+   * @throws IOException
+   */
+  public MigrationZipFile(Path zipPath, Path keyPath, Path checksumPath) throws IOException {
+    Validate.notNull(zipPath, "Null zip path");
+    Validate.notNull(keyPath, "Null key path");
+    Validate.notNull(checksumPath, "Null checksum path");
+    if ((!zipPath.toFile().exists())
+        || (!keyPath.toFile().exists())
+        || (!checksumPath.toFile().exists())) {
+      throw new FileNotFoundException(
+          String.format(
+              "Could not find all required files [%s], [%s], [%s]",
+              zipPath, keyPath, checksumPath));
+    }
+    this.zipPath = zipPath;
+    this.keyPath = keyPath;
+    this.checksumPath = checksumPath;
+    checksumValid = validateChecksum(zipPath, checksumPath);
+    zipFile = new ZipFile(zipPath.toFile());
+    cipher = initCipher();
+  }
+
+  /**
+   * Wraps the {@link InputStream} from ZipFile.getInputStream with a {@link CipherInputStream}
+   *
+   * @param entry
+   * @return
+   * @throws IOException
+   */
+  public InputStream getInputStream(ZipEntry entry) throws IOException {
+    return getCipherInputStreamFor(new BufferedInputStream(zipFile.getInputStream(entry)));
+  }
+
+  /**
+   * Proxy for ZipFile.stream();
+   *
+   * @return
+   */
+  @SuppressWarnings(
+      "squid:S1452" /* Return type needs to use generic wildcard as this method proxies an existing one from ZipFile */)
+  public Stream<? extends ZipEntry> stream() {
+    return zipFile.stream();
+  }
+
+  /**
+   * Determines if the MigrationZipFile checksum is valid.
+   *
+   * @return the boolean corresponding the checksum validity
+   */
+  public boolean isValidChecksum() {
+    return checksumValid;
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (!closed) {
+      closed = true;
+      zipFile.close();
+    }
+  }
+
+  public Path getZipPath() {
+    return zipPath;
+  }
+
+  public Path getChecksumPath() {
+    return checksumPath;
+  }
+
+  public Path getKeyPath() {
+    return keyPath;
+  }
+
+  public boolean isClosed() {
+    return closed;
+  }
+
+  /**
+   * Closes the open {@link ZipFile} and deletes the file, key, and checksum
+   *
+   * @return
+   */
+  @SuppressWarnings("squid:S2178" /* using & to ensure all files are deleted before return */)
+  public boolean deleteQuitetly() {
```

Comment:
❗️ 1. Really dislike the abuse of the bitwise operator here. It _works_, but at the cost of using a construct that's fallen far out of favor.
2. And...as you're never using the value returned from this method, there's little value in calculating it.

Reply:
Updated this to be a `void` method and removed the bitwise operators and the related suppression

Reply:
I am good with changing it to void. 

I wouldn't call the use of `&` versus `&&` as an abuse. It is a standard technique for doing what was required here without complicating the code.